test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py

```diff
@@ -620,3 +620,47 @@ def test_pytorch_standalone_hpu(
         container_name="ec2_training_habana_pytorch_container",
         enable_habana_async_execution=True,
     )
+
+
+@pytest.mark.usefixtures("feature_aws_framework_present")
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.integration("cudnn")
+@pytest.mark.model("N/A")
+@pytest.mark.parametrize("ec2_instance_type", PT_EC2_SINGLE_GPU_INSTANCE_TYPE, indirect=True)
+def test_pytorch_cudnn_match_gpu(
+    pytorch_training, ec2_connection, region, gpu_only, ec2_instance_type, pt21_and_above_only
+):
+    """
+    PT 2.1 reintroduces a dependency on CUDNN to support NVDA TransformerEngine. This test is to ensure that torch CUDNN matches system CUDNN in the container.
+    """
+    container_name = "pt_cudnn_test"
+    ec2_connection.run(f"$(aws ecr get-login --no-include-email --region {region})", hide=True)
+    ec2_connection.run(f"docker pull -q {pytorch_training}", hide=True)
+    ec2_connection.run(
+        f"nvidia-docker run --name {container_name} -itd {pytorch_training}", hide=True
+    )
+    major_cmd = "cat /usr/include/cudnn_version.h | grep '#define CUDNN_MAJOR'"
+    minor_cmd = "cat /usr/include/cudnn_version.h | grep '#define CUDNN_MINOR'"
+    patch_cmd = "cat /usr/include/cudnn_version.h | grep '#define CUDNN_PATCHLEVEL'"
+    major = ec2_connection.run(
+        f"nvidia-docker exec --user root {container_name} bash -c '{major_cmd}'", hide=True
+    ).stdout.split()[-1]
+    minor = ec2_connection.run(
+        f"nvidia-docker exec --user root {container_name} bash -c '{minor_cmd}'", hide=True
+    ).stdout.split()[-1]
+    patch = ec2_connection.run(
+        f"nvidia-docker exec --user root {container_name} bash -c '{patch_cmd}'", hide=True
+    ).stdout.split()[-1]
+
+    cudnn_from_torch = ec2_connection.run(
+        f"nvidia-docker exec --user root {container_name} python -c 'from torch.backends import cudnn; print(cudnn.version())'",
```

Comment:
this cudnn comes from pytorch and not from installed from OS package, right?

Reply:
This cudnn represents the cudnn version that torch is compiled with, not the DLC cudnn version. There are basically static links to cudnn from torch - while it doesn't appear to be a big issue if there are slightly different versions of cudnn from compile --> system, adding this test for future safety so that the versions don't go out of sync